react/features/toolbox/components/web/Toolbox.tsx

```diff
@@ -1011,6 +1030,11 @@ class Toolbox extends Component<IProps> {
             !_jwtDisabledButons.includes(key)
             && (isToolbarButtonEnabled(key, _toolbarButtons) || isToolbarButtonEnabled(alias, _toolbarButtons))
         );
+        const filteredKeys = filtered.map(button => button.key);
```

Comment:
Why is this logic necessary?

Reply:
In order to handle the scenario where a button defined in `order` turns out to be disabled. In this case we need to calculate how much buttons we need to display on the main toolbar. If we go with the previous implementation some random buttons from the overflow menu are actually displayed on the main toolbar.

Reply:
I see. Isn't that a separate change from this though?

Reply:
It could be. That was existing problem even before. I will move it in a separate commit.